bitcoin/script/miniscript.h

```diff
@@ -807,45 +807,50 @@ static constexpr int MAX_PARSE_RECURSION = 201;
 
 //! Parse a miniscript from its textual descriptor form.
 template<typename Key, typename Ctx>
-inline NodeRef<Key> Parse(Span<const char>& in, const Ctx& ctx, int recursion_depth) {
+inline NodeRef<Key> Parse(Span<const char>& in, const Ctx& ctx, int recursion_depth, bool wrappers_parsed = false) {
     if (recursion_depth >= MAX_PARSE_RECURSION) {
         return {};
     }
     auto expr = Expr(in);
     // Parse wrappers
-    for (int i = 0; i < expr.size(); ++i) {
-        if (expr[i] == ':') {
-            auto in2 = expr.subspan(i + 1);
-            auto sub = Parse<Key>(in2, ctx, recursion_depth + 1);
-            if (!sub || in2.size()) return {};
-            for (int j = i; j-- > 0; ) {
-                if (expr[j] == 'a') {
-                    sub = MakeNodeRef<Key>(NodeType::WRAP_A, Vector(std::move(sub)));
-                } else if (expr[j] == 's') {
-                    sub = MakeNodeRef<Key>(NodeType::WRAP_S, Vector(std::move(sub)));
-                } else if (expr[j] == 'c') {
-                    sub = MakeNodeRef<Key>(NodeType::WRAP_C, Vector(std::move(sub)));
-                } else if (expr[j] == 'd') {
-                    sub = MakeNodeRef<Key>(NodeType::WRAP_D, Vector(std::move(sub)));
-                } else if (expr[j] == 'j') {
-                    sub = MakeNodeRef<Key>(NodeType::WRAP_J, Vector(std::move(sub)));
-                } else if (expr[j] == 'n') {
-                    sub = MakeNodeRef<Key>(NodeType::WRAP_N, Vector(std::move(sub)));
-                } else if (expr[j] == 'v') {
-                    sub = MakeNodeRef<Key>(NodeType::WRAP_V, Vector(std::move(sub)));
-                } else if (expr[j] == 't') {
-                    sub = MakeNodeRef<Key>(NodeType::AND_V, Vector(std::move(sub), MakeNodeRef<Key>(NodeType::JUST_1)));
-                } else if (expr[j] == 'u') {
-                    sub = MakeNodeRef<Key>(NodeType::OR_I, Vector(std::move(sub), MakeNodeRef<Key>(NodeType::JUST_0)));
-                } else if (expr[j] == 'l') {
-                    sub = MakeNodeRef<Key>(NodeType::OR_I, Vector(MakeNodeRef<Key>(NodeType::JUST_0), std::move(sub)));
-                } else {
-                    return {};
+    if (!wrappers_parsed) {
+        for (int i = 0; i < expr.size(); ++i) {
```

Comment:
Instead of checking `i != 0` below, could you instead start this loop at `i = 1`?

Reply:
Yes, I can do it. And that is indeed much cleaner.